ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -0,0 +1,207 @@
+import {Button, Clickable, MenuItem} from 'app/components/buttons';
+import {FlexColumn, FlexRow} from 'app/components/flex';
+import {ClrIcon} from 'app/components/icons';
+import {PopupTrigger} from 'app/components/popups';
+import {Spinner} from 'app/components/spinners';
+import {runtimeApi} from 'app/services/swagger-fetch-clients';
+import colors, {addOpacity} from 'app/styles/colors';
+import {reactStyles, withCurrentWorkspace} from 'app/utils';
+import {machineTypes} from 'app/utils/machines';
+import {WorkspaceData} from 'app/utils/workspace-data';
+import {Dropdown} from 'primereact/dropdown';
+import {InputNumber} from 'primereact/inputnumber';
+
+import {Runtime} from 'generated/fetch';
+
+import * as fp from 'lodash/fp';
+import * as React from 'react';
+
+
+const styles = reactStyles({
+  sectionHeader: {
+    color: colors.primary,
+    fontSize: '16px',
+    fontWeight: 700,
+    lineHeight: '1rem',
+    marginBottom: '12px',
+    marginTop: '12px'
+  },
+  controlSection: {
+    backgroundColor: String(addOpacity(colors.white, .75)),
+    borderRadius: '3px',
+    padding: '.75rem',
+    marginTop: '.75rem'
+  },
+  presetMenuItem: {
+    color: colors.primary,
+    fontSize: '14px'
+  }
+});
+
+const defaultMachineType = machineTypes.find(({name}) => name === 'n1-standard-4');
+
+export interface Props {
+  workspace: WorkspaceData;
+}
+
+interface State {
+  // Whether the initial runtime load is still in progress.
+  loading: boolean;
+  // Whether there was an error in loading the runtime data.
+  error: boolean;
+  // The runtime. null if none exists, or if there was an error in loading the
+  // runtime.
+  runtime: Runtime|null;
+}
+
+export const RuntimePanel = withCurrentWorkspace(
```

Comment:
I had to add `()` to get it to compile
```suggestion
export const RuntimePanel = withCurrentWorkspace()(
```